torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1164,10 +1164,40 @@ def _register_param_handles_from_handles(
         for flat_param in self.params:
             self._init_param_attributes(flat_param)
         self._register_flat_params()
+        self._construct_module_to_handles()
         # Re-register the forward hooks since the handle construction changed
         self._register_pre_forward_hooks()
         self._register_post_forward_hooks()
 
+    def _construct_module_to_handles(self) -> None:
+        """
+        This constructs ``_module_to_handles`` based on ``_handles``. For each
+        `handle`, its `module` is chosen as the lowest common ancestor that
+        includes all unflatten parameters included in this `handle.`
+        """
+        module_to_parent : Dict[nn.Module, nn.Module] = dict()
+        for module in self.module.modules():
+            for child in module.children():
+                module_to_parent[child] = module
+        for handle in self._handles:
+            roots = list(handle.flat_param._root_modules)
+            assert (
```

Comment:
```suggestion
            p_assert (
```
since this may be called in the post-backward callback (from `_register_param_handles_from_handles()`)